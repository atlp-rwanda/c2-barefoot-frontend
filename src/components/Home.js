import React, { Component } from 'react'
import Button from '@material-ui/core/Button'
import '../App.css'
import { increment, decrement } from '../actions/counterAction'
import { connect } from 'react-redux'

class Home extends Component{

    handleIncrement =  () => {
        this.props.increment()
    }

    handleDecrement =  () => {
        this.props.decrement()
    }

    render(){
        return(
            <div className="home" >
                 <Button variant="outlined" color='primary' onClick={this.handleDecrement}>-</Button>
                <div>Counter: {this.props.count}</div>
                 <Button variant="outlined" color='primary' onClick={this.handleIncrement}>+</Button>
            </div>
        )
    }
}

function mapDispatchToProps(dispatch){
    return{
        increment: () => dispatch(increment()),
        decrement: () => dispatch(decrement())
    }
}

function mapStateToProps(state){
    return({
        count: state.counter
    })
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);