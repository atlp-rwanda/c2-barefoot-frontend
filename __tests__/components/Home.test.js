import React from 'react';
import Home from '../../src/components/Home';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/redux/store';
describe('<Home />', ()=>{
    it('matches the snapshot', ()=>{
        let hometest = (<Provider store={store}><Home/></Provider>);
        let tree = renderer.create(hometest).toJSON();
        expect(tree).toMatchSnapshot();
    })
})