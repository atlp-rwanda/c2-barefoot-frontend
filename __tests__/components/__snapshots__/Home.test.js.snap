// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
<div
  className="home"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      -
    </span>
  </button>
  <div>
    Counter: 
    0
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      +
    </span>
  </button>
</div>
`;
